litespi/spi_nor_flash_module.py

```diff
@@ -180,113 +189,68 @@ class SpiNorFlashModule(metaclass=MetaSizes):
         setup module to run in DDR mode
     """
 
-    def _configure_transfer_mode(self, cfg):
+    def _configure_transfer_mode(self, read_cmd):
         """Configure Transfer Mode
 
         Funtion ``_configure_transfer_mode`` setups fields:
 
-            * addr_bits
+            * cmd_width
             * addr_width
             * bus_width
+            * addr_bits
+            * ddr
 
-        according to user defined config and checks if only one
-        mode has been selected.
+        according to user defined read command.
 
         Arguments
         ---------
-        cfg : SpiNorFeatures
-            User defined flags used to configure chip into desired mode.
-
-        Return value
-        ------------
-        bool
-            True: correct chip configuration
-            False: more than one mode selected (incorrect configuration)
+        read_cmd : SpiNorFlashOpCode
+            User defined read opcode used to configure chip into desired mode.
         """
 
-        if SpiNorFeatures.FEATURE_4BA in cfg:
+        cmd_name = SpiNorFlashOpCodes.get_name_from_value(SpiNorFlashOpCodes, read_cmd)
```

Comment:
This is actually pretty clever! This code should be functionality (probably @property) of the SpiNorFlashOpCodes class.

```python
class SpiNorFlashOpCodes:
 @property
 def _configs(self):
   return self.name.split("_")[1:]

 @property
 def cmd_width(self):
   self._configs[0]
```
IE `cmd_name.cmd_width`, `cmd_name.addr_bits`, etc

Reply:
I suggest to do it like this: [link](https://github.com/litex-hub/litespi/pull/22/files#diff-077a045c4d22989e305fdff24affcea3R165-R186)